cmake/HalideGeneratorHelpers.cmake

```diff
@@ -608,7 +608,16 @@ function(add_halide_runtime RT)
         _Halide_fix_xcode("${RT}")
     endif ()
 
-    target_link_libraries("${RT}" INTERFACE Halide::Runtime Threads::Threads ${CMAKE_DL_LIBS})
+    target_link_libraries("${RT}" INTERFACE Halide::Runtime)
+    if (NOT TARGET Threads::Threads)
+        set(THREADS_PREFER_PTHREAD_FLAG TRUE)
+        find_package(Threads QUIET)
+    endif()
+    if (TARGET Threads::Threads)
+        target_link_libraries("${RT}" INTERFACE Threads::Threads ${CMAKE_DL_LIBS})
+    else()
+        message(STATUS "Package Threads not found, skip linking it to Halide::Runtime")
```

Comment:
This is going to search for threads on every invocation if it is not present. Thus this message will print once per call and be quite noisy. Can we change this to write to a global property like `Halide_RUNTIME_NO_THREADS` that can be checked to avoid the search and warning?

Also, I'm not sure why `${CMAKE_DL_LIBS}` is only applied when `Threads::Threads` is around. Shouldn't it always be included?